Extensions/render/backend/port.lisp

```diff
@@ -3,24 +3,30 @@
 ;;; Port
 
 (defclass render-port-mixin (basic-port)
-  ((all-font-families :initform nil :accessor all-font-families)))
+  ((all-font-families :initform nil :accessor all-font-families)
+   (mirror->%image :initform (make-hash-table))))
 
 (defun mirror->%image (port mirror)
-  (declare (ignore port))
-  mirror)
+  (gethash mirror (slot-value port 'mirror->%image)))
+
+(defun (setf mirror->%image) (%image port mirror)
+  (check-type %image (or null image-mirror-mixin))
+  (if %image
+      (setf (gethash mirror (slot-value port 'mirror->%image)) %image)
+      (remhash mirror (slot-value port 'mirror->%image))))
 
 ;;; change geometry
 
 (defmethod port-set-mirror-region :after ((port render-port-mixin) mirror region)
-  (let ((sheet (port-lookup-sheet port mirror)))
-    (%set-image-region (mirror->%image port (sheet-mirror sheet)) region)))
+  (%set-image-region (mirror->%image port mirror) region))
 
 (defmethod port-set-mirror-transformation :after ((port render-port-mixin) mirror transformation)
   (declare (ignore port mirror transformation))
   nil)
 
 ;;; realize/destroy mirrors
-(defmethod realize-mirror ((port render-port-mixin) (sheet image-sheet-mixin)))
+(defmethod realize-mirror ((port render-port-mixin) (sheet image-sheet-mixin))
+  (gensym "RENDER-PORT-MIRROR"))
```

Comment:
Please explain this in a comment.

Reply:
Added a comment

```
;;; We return a gensym because the RENDER-PORT-MIXIN maintains a hash table
;;; which maps mirrors to images - a key must be unique. -- jd 2020-11-09
```

that said, there is other reason that is not worth mentioning in the comment because it applies to all realize-mirror functions - realize-mirror returns a sheet mirror, and NIL is described as a value distinct for sheets that do not have a mirror. That applies to all realize-mirror methods (the mirror could be a single symbol, i.e 'mirror, but not nil).

fixed.